src/components/organisms/NavBarSchedule/utils.ts

```diff
@@ -0,0 +1,38 @@
+import {
+  differenceInMinutes,
+  endOfDay,
+  getUnixTime,
+  max,
+  min,
+  startOfDay,
+} from "date-fns";
+
+import { PersonalizedVenueEvent, VenueEvent } from "types/venues";
+import { MyPersonalizedSchedule } from "types/User";
+
+import { WithVenueId } from "utils/id";
+import { eventEndTime, eventStartTime } from "utils/event";
+import { isTruthy } from "utils/types";
+
+export const prepareForSchedule = (
+  day: Date,
+  usersEvents: MyPersonalizedSchedule
+) => (event: WithVenueId<VenueEvent>): PersonalizedVenueEvent => {
+  const startOfEventToShow = max([eventStartTime(event), startOfDay(day)]);
+  const endOfEventToShow = min([eventEndTime(event), endOfDay(day)]);
+
+  return {
+    ...event,
+    start_utc_seconds: getUnixTime(startOfEventToShow),
+    duration_minutes: differenceInMinutes(endOfEventToShow, startOfEventToShow),
+    isSaved: isTruthy(
+      event.id && usersEvents[event.venueId]?.includes(event.id)
+    ),
```

Comment:
I did that like you said here: https://github.com/sparkletown/sparkle/pull/1426/commits/6ff0df273fef51c6b69a313b4be36fa82d94a699

@0xdevalias but now can you explain why don't you like `isTruthy()` function? you seem to avoid it. So when should I use it then? 
I thought that here's a good example of its usage cause we don't need to do this `?? false : false` thing which doesn't look nice to my eye, and `isTruthy()` could do all the work for us.

Reply:
> can you explain why don't you like `isTruthy()` function? you seem to avoid it. So when should I use it then?

It was a function I made to try and replace our usage of `!!` to coerce things into being `boolean` in a cleaner way (and fix some related TypeScript nuances about it not understanding that that would imply that the variable is defined)

My main issue/aversion with it is that it's a very 'blunt'/'crude' tool, and can lead to weird issues (eg. `0` isn't truthy, so it will return `False`; yet often `0` is a valid defined value). So where we can be more explicit/exact in our checks, we should do so.

In many cases `isDefined` (aka: `notEmpty`, `src/utils/types.ts`) is a better/more explicit version that can be used instead of `isTruthy`

In this case, the ternary version I suggested probably isn't great since it has to specify `false` twice, but the first suggestion is probably cleaner/better and likely matches what we do across the codebase for this more often.

We could likely extract this pattern, since it seems to come up a fair bit, something like:

`src/utils/types.ts` (or somewhere better if you can think of it)
```typescript
export const arrayIncludes = <T>(arr: T[], element?: T | null): boolean => {
  if (!isDefined(element)) return false

  return arr.includes(element)
}
```

Reply:
thanks for the explanation. Will use `isDefined` then when needed.
and created the function here: https://github.com/sparkletown/sparkle/pull/1426/commits/34e272296e8ffbdcf9ecdcc4134e3620eafb1ecf